enhancements/operator-conditions-lib.md

```diff
@@ -49,40 +49,99 @@ Operators need to be able to read the Status CR created by OLM, and also have th
 
 #### Story 1
 
-As an operator author, I want my operator to be able to add/modify conditions in the `Status` CR.
-
-#### Story 2
-
-As an operator author, I would like to know the `NamespacedName` of the `Status` CR, which can be used to obtain the resource from cluster.
+As an operator author, I want my operator to be able to get/update conditions in the `OperatorCondition` CR.
 
 ## Design Details
 
-When an operator is installed in the cluster, OLM will create a `Status` Custom Resource for the operator. The operator will be able to perform the following actions using the APIs provided in the library:
+When an operator is installed in the cluster, OLM will create an `OperatorCondition` Custom Resource for the operator. The user will be expected to provide a [client][cr_client] using which the library can access the operator condition resource present in the cluster, owned by the operator. The library will provide an interface, using which users can get the required condition from the CRD, or can set the status of the specific condition.
+
+There would be a generic operatorCondition manager interface, which would in turn have the methods to `get` and `set` a [conditionType][condition_type_api]. For example:
 
-1. Get the `NamespacedName` of the CRD:
+```go
+type ConditionManager interface {
+  // GetCondition gets the condition on the operator's 
+	// OperatorCondition. It returns an error if there are problems getting
+	// the OperatorCondition object or if the specific condition type does not
+	// exist.
+  GetCondition(context.Context, metav1.ConditionType) (*metav1.Condition, error)
+
+  // SetConditionStatus sets the specific condition on the operator's 
+	// OperatorCondition to the provided status. It returns an error if there
+	// are problems getting or updating the OperatorCondition object.
+  SetConditionStatus(context.Context, metav1.ConditionType, metav1.ConditionStatus, ...Option) error
```

Comment:
The only downside I see with this approach is the `Conditions` interface becomes non-composable because you can't override method names in Go, ex. you can't have two `Get`s with different signatures in the same interface. That isn't an immediate problem though, so I think these simpler method names should be adopted.